### PR TITLE
[PAL] Replace `UNKNOWN_HANDLE` macro with inline code

### PR DESCRIPTION
**Pull request by billionairiam**
Signed-off-by: Liang Ma <liang3.ma@intel.com>

## Description of the changes 
Replace `UNKNOW_HANDLE` macros with inline code.

---
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/gramine/775)

### CODE REVIEW

**Review by billionairiam — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/775)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel), period found at the end of a one-liner (waiting on @dimakuv)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/gramine/775#-N7UXnUf0giDUn9gtrEs):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

- `UNKNOW` -> `UNKNOWN` (typo)

- `macros` -> `macro` (singular)

- No dot at the end

We will fix it during final rebase.
</blockquote></details>

Done.

**Review by boryspoplawski — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/775)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required), period found at the end of a one-liner (waiting on @dimakuv)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/gramine/775#-N7UXnUf0giDUn9gtrEs):*
<details><summary><i>Previously, billionairiam (Liang Ma) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

And also wrap `UNKNOWN_HANDLE` in backticks.

@billionairiam Please do not respond `Done.` to things that are not in fact done. This will be fixed during final rebase.

**Comment by boryspoplawski** [comment]
Jenkins, test this please

**Review by dimakuv — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/775)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL), period found at the end of a one-liner (waiting on @billionairiam and @dimakuv)

___
*[`pal/src/pal_object.c` line 51 at r1](https://reviewable.io/reviews/gramineproject/gramine/775#-N7VZhm277wAZ2UxN7Ed:-N7VZhm277wAZ2UxN7Ee:b-w225li) ([raw file](https://github.com/gramineproject/gramine/blob/3942026a802ba8148bb9265a21f51ea09f07e16a/pal/src/pal_object.c#L51)):*
> ```C
>     for (size_t i = 0; i < count; i++) {
>         assert(handle_array[i]);
>         assert(!handle_array[i]->hdr.type >= PAL_HANDLE_TYPE_BOUND);
> ```

Oops, we didn't notice the bug here, but our CI noticed it:
```
12:21:55  In file included from ../pal/include/arch/x86_64/pal_arch.h:18:0,
12:21:55                   from ../pal/include/pal/pal.h:68,
12:21:55                   from ../pal/src/pal_object.c:8:
12:21:55  ../pal/src/pal_object.c: In function ‘PalStreamsWaitEvents’:
12:21:55  ../pal/src/pal_object.c:51:43: error: comparison of constant ‘11’ with boolean expression is always false [-Werror=bool-compare]
12:21:55           assert(!handle_array[i]->hdr.type >= PAL_HANDLE_TYPE_BOUND);
```

Basically, this is because of the `!` here. We should have added `()` around the check. But now that I'm looking at it, let's just switch the check like this:
```
        assert(handle_array[i]->hdr.type < PAL_HANDLE_TYPE_BOUND);
```

**Review by mkow — CHANGES_REQUESTED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/775)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: ITL), period found at the end of a one-liner (waiting on @billionairiam and @dimakuv)

**Review by billionairiam — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/775)* status: 2 of 3 files reviewed, 2 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ), period found at the end of a one-liner, "fixup! " found in commit messages' one-liners (waiting on @boryspoplawski, @dimakuv, and @mkow)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/gramine/775#-N7UXnUf0giDUn9gtrEs):*
<details><summary><i>Previously, boryspoplawski (Borys Popławski) wrote…</i></summary><blockquote>

And also wrap `UNKNOWN_HANDLE` in backticks.

@billionairiam Please do not respond `Done.` to things that are not in fact done. This will be fixed during final rebase.
</blockquote></details>

Get it!
___
*[`pal/src/pal_object.c` line 51 at r1](https://reviewable.io/reviews/gramineproject/gramine/775#-N7VZhm277wAZ2UxN7Ed:-N7YWYb01keaywleicN9:b-fkjt46) ([raw file](https://github.com/gramineproject/gramine/blob/3942026a802ba8148bb9265a21f51ea09f07e16a/pal/src/pal_object.c#L51)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Oops, we didn't notice the bug here, but our CI noticed it:
```
12:21:55  In file included from ../pal/include/arch/x86_64/pal_arch.h:18:0,
12:21:55                   from ../pal/include/pal/pal.h:68,
12:21:55                   from ../pal/src/pal_object.c:8:
12:21:55  ../pal/src/pal_object.c: In function ‘PalStreamsWaitEvents’:
12:21:55  ../pal/src/pal_object.c:51:43: error: comparison of constant ‘11’ with boolean expression is always false [-Werror=bool-compare]
12:21:55           assert(!handle_array[i]->hdr.type >= PAL_HANDLE_TYPE_BOUND);
```

Basically, this is because of the `!` here. We should have added `()` around the check. But now that I'm looking at it, let's just switch the check like this:
```
        assert(handle_array[i]->hdr.type < PAL_HANDLE_TYPE_BOUND);
```
</blockquote></details>

Already changed!

**Review by dimakuv — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/775)* status: all files reviewed, 2 unresolved discussions, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel), period found at the end of a one-liner, "fixup! " found in commit messages' one-liners (waiting on @dimakuv and @mkow)

**Comment by dimakuv** [comment]
Jenkins, test this please

**Review by mkow — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/775)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (1 more required), period found at the end of a one-liner, "fixup! " found in commit messages' one-liners (waiting on @dimakuv)

**Review by boryspoplawski — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/775)* status: all files reviewed, 1 unresolved discussion, period found at the end of a one-liner, "fixup! " found in commit messages' one-liners (waiting on @dimakuv)

**Review by dimakuv — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/775)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel) (waiting on @boryspoplawski)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/gramine/775#-N7UXnUf0giDUn9gtrEs):*
<details><summary><i>Previously, billionairiam (Liang Ma) wrote…</i></summary><blockquote>

Get it!
</blockquote></details>

Done

**Comment by dimakuv** [comment]
Jenkins, test this please

**Review by boryspoplawski — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/775)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (1 more required)

**Review by mkow — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/gramine/775)* status: :shipit: complete! all files reviewed, all discussions resolved
